ykcapi/src/lib.rs

```diff
@@ -48,6 +52,116 @@ pub extern "C" fn __ykutil_get_llvmbc_section(res_addr: *mut *const c_void, res_
     }
 }
 
+struct Registers {
+    rax: usize,
+    rdx: usize,
+    rcx: usize,
+    rbx: usize,
+    rsi: usize,
+    rdi: usize,
+    rsp: usize,
+}
+
+impl Registers {
+    fn from_ptr(ptr: *const c_void) -> Registers {
+        Registers {
```

Comment:
Are these the caller-save registers defined by the ABI?

Reply:
Actually, this is supposed to be all possible registers, but I just realised we are missing a few. Whoops.

Reply:
OK, but what's not clear to me:

Later we do:
```
let registers = Registers::from_ptr(rsp);
```

and `from_ptr()` loads the registers from the stack. Why would *all* registers have been spilled at this point? Does the deoptimise intrinsic implicitly do that for us?

Reply:
I spilled them manually at the beginning of the `__llvm_deoptimize` call: https://github.com/ykjit/yk/pull/460/files#diff-65ddb41c4a2e6cf6d0de3ba8be066c3ac59faa885f4f52b065b5c35b8352697eR65

Reply:
I think it's time to refactor this struct a bit so we don't explicitly name all registers.

Reply:
Partly fixed in efcb294.

I've decided against adding more registers since we currently don't have tests that use them. Once we do we'll hit an `unreachable!` so we'll know to add them.